h/presenters/group_json.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+
+class GroupJSONPresenter(object):
+    """Present a group in the JSON format returned by API requests."""
+
+    def __init__(self, group, route_url=None):
+        self.group = group
+        self._route_url = route_url
+
+    def asdict(self):
+        return self._model(self.group)
+
+    def _model(self, group):
+        model = {
+          'name': group.name,
+          'id': group.pubid,
+          'public': group.is_public,
+          'scoped': False,  # TODO
+          'type': 'open' if group.is_public else 'private'  # TODO
+        }
+        model = self._inject_urls(group, model)
+        return model
+
+    def _inject_urls(self, group, model):
+        model['urls'] = {}
+        if not self._route_url:
+            return model
+
+        model['url'] = self._route_url('group_read',
+                                       pubid=group.pubid,
+                                       slug=group.slug)
+        model['urls']['group'] = model['url']
+        return model
+
+
+class GroupsJSONPresenter(GroupJSONPresenter):
+    """Present a list of groups as JSON"""
+
+    def __init__(self, groups, route_url=None):
+        self.groups = groups
+        self._route_url = route_url
+
+    def asdicts(self):
+        return [self._model(group) for group in self.groups]
```

Comment:
Now that I think of it inheritance is probably not the best thing here, because `GroupJSONPresenter` has a `group` property that, in the case of a `GroupsJSONPresenter`, never gets created. Just having `GroupsJSONPresenter` _use_ `GroupJSONPresenter` instead of inherit from it might be cleaner, e.g. `return [GroupJSONPresenter(group, self._route_url).asdict() for group in self.groups]`.

Or make `_model()` and `_inject_urls()` module-level functions then `GroupJSONPresenter` and `GroupsJSONPresenter` could be entirely independent classes that both call those functions.

But this way is fine though, I'm happy to merge it as-is

Reply:
I'd like to fix this in a subsequent PR as it is rather silly as-is.